From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nomorecoolnicknames <nomorecoolnicknames@users.noreply.github.com>
Date: Tue, 28 Jul 2026 13:42:50 +0300
Subject: [PATCH 4/5] rild: keep the multi-SIM cflags, fix the build

LOCAL_CFLAGS := -DRIL_SHLIB discarded the -DANDROID_MULTI_SIM and
-DANDROID_SIM_COUNT_2 set just above, while libril is always built with
them. ANDROID_MULTI_SIM changes the RIL_Env callbacks, so rild and libril
disagreed on the layout of the same struct. Append instead.

Also link libutils, add the libcutils include path, and define LOG_TAG
before the RIL header.
---
 ril/rild/Android.mk | 8 ++++++--
 ril/rild/rild.c     | 2 +-
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/ril/rild/Android.mk b/ril/rild/Android.mk
index 598773e..04e470b 100644
--- a/ril/rild/Android.mk
+++ b/ril/rild/Android.mk
@@ -39,12 +39,17 @@ LOCAL_SRC_FILES:= \
 LOCAL_SHARED_LIBRARIES := \
 	liblog \
 	libcutils \
+	libutils \
 	libril \
 	libdl
 
 LOCAL_STATIC_LIBRARIES := \
 	rild-prop-md1
 
+LOCAL_C_INCLUDES += \
+	system/core/libcutils/include
+
+
 LOCAL_CFLAGS += -DANDROID_MULTI_SIM
 LOCAL_CFLAGS += -DANDROID_SIM_COUNT_2
 
@@ -52,7 +57,7 @@ LOCAL_CFLAGS += -DANDROID_SIM_COUNT_2
 LOCAL_WHOLE_STATIC_LIBRARIES := \
 	librilutils_static
 
-LOCAL_CFLAGS := -DRIL_SHLIB
+LOCAL_CFLAGS += -DRIL_SHLIB
 
 LOCAL_MODULE_RELATIVE_PATH := hw
 LOCAL_PROPRIETARY_MODULE := true
@@ -61,4 +66,3 @@ LOCAL_MODULE:= rild
 
 include $(BUILD_EXECUTABLE)
 endif
-
diff --git a/ril/rild/rild.c b/ril/rild/rild.c
index aac4f0d..4c23f93 100644
--- a/ril/rild/rild.c
+++ b/ril/rild/rild.c
@@ -27,8 +27,8 @@
 #include <fcntl.h>
 #include <errno.h>
 
-#include <telephony/ril.h>
 #define LOG_TAG "RILD"
+#include <telephony/ril.h>
 #include <log/log.h>
 #include <cutils/properties.h>
 #include <cutils/sockets.h>
